collectors/apicmcpfault.py

```diff
@@ -0,0 +1,58 @@
+import logging, re
+
+from prometheus_client.core import CounterMetricFamily, Summary
+import BaseCollector
+
+LOG = logging.getLogger('apic_exporter.exporter')
+REQUEST_TIME = Summary('network_apic_mcp_fault_counter',
+                       'Time spent processing request')
+
+
+class ApicMCPCollector(BaseCollector.BaseCollector):
+    def describe(self):
+        yield CounterMetricFamily('network_apic_mcp_fault_counter',
+                                  'Counter for MCP Faults')
+
+    @REQUEST_TIME.time()
+    def collect(self):
+        LOG.debug('Collecting APIC MCP Fault metrics ...')
+
+        c_mcp_faults = CounterMetricFamily(
+            'network_apic_mcp_fault_counter',
+            'Counter for MCP Faults',
+            labels=['apicHost', 'ip', 'fault_lifecyle', 'fault_summary', 'fault_desc'])
+
+        metric_counter = 0
+        query = "/api/node/class/faultInst.json?query-target-filter=or(eq(faultInst.code,\"F2533\"),eq(faultInst.code,\"F2534\"))"
+        for host in self.hosts: 
+            fetched_data = self.connection.getRequest(host, query)
+            if not self.connection.isDataValid(fetched_data):
+                LOG.warning(
+                    "Skipping apic host %s, %s did not return anything", host,
+                    query)
+                continue
+
+            if len(fetched_data['imdata']) == 0:
+                output = json.loads(query.text)
+                # Add Empty Counter to have the metric show up in Prometheus.
+                # Otherwise they only show when something is wrong and we dont know if it is actually working
+                c_mcp_faults.add_metric(labels=[host, '', '', ''], value=0)
```

Comment:
```suggestion
                c_mcp_faults.add_metric(labels=[host, '', '', ''], 0)
```

no kwarg according to:
https://github.com/prometheus/client_python/blob/e92fa050090cb99bc724a8e4045c46b5096cbbd3/prometheus_client/metrics_core.py#L86